march_shared_classes/src/march_shared_classes/gait/subgait.py

```diff
@@ -301,3 +330,31 @@ def __getitem__(self, index):
     def __len__(self):
         return len(self.joints)
     # endregion
+
+    @staticmethod
+    def validate_version(gait_path, subgait_name, version):
+        """Check whether a gait exists for the gait."""
+        if version.startswith(PARAMETRIC_GAITS_PREFIX):
+            base_version, other_version, _ = Subgait.unpack_parametric_version(version)
+            base_subgait_path = os.path.join(gait_path, subgait_name, base_version + '.subgait')
+            other_subgait_path = os.path.join(gait_path, subgait_name, other_version + '.subgait')
+            for subgait_path in [base_subgait_path, other_subgait_path]:
+                if not os.path.isfile(subgait_path):
+                    rospy.logwarn('{sp} does not exist'.format(sp=subgait_path))
+                    return False
+        else:
+            subgait_path = os.path.join(gait_path, subgait_name, version + '.subgait')
+            if not os.path.isfile(subgait_path):
+                rospy.logwarn('{sp} does not exist'.format(sp=subgait_path))
+                return False
+        return True
+
+    @staticmethod
+    def unpack_parametric_version(version):
+        """Unpack a version to base version, other version and parameter."""
+        parameter_str = re.search(r'{0}[0-9.]*_'.format(PARAMETRIC_GAITS_PREFIX), version).group(0)
+        parameter = float(parameter_str[1:-1])
```

Comment:
Because the prefix can now be multiple characters, it is nicer to match a group instead of matching the entire string and getting a substring starting from 1. Furthermore, the float regex could be empty. I replaced it with another regex that assumes you don't use negative parameters and doesn't allow floats like 0. and .0
```suggestion
        parameter_str = re.search(r'^{0}(\d+\.\d+)_'.format(PARAMETRIC_GAITS_PREFIX), version).group(1)
        parameter = float(parameter_str)
```

Reply:
Thanks I don't know regex that well.